tests/components/transmission/conftest.py

```diff
@@ -0,0 +1,229 @@
+"""Fixtures for the Transmission integration tests."""
+import json
+from unittest.mock import patch
+
+import pytest
+from transmissionrpc.error import TransmissionError
+
+from homeassistant.components.transmission import (
+    CONF_HOST,
+    CONF_LIMIT,
+    CONF_NAME,
+    CONF_ORDER,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    DOMAIN,
+)
+from homeassistant.components.transmission.const import (
+    ORDER_BEST_RATIO_FIRST,
+    ORDER_NEWEST_FIRST,
+    ORDER_OLDEST_FIRST,
+    ORDER_WORST_RATIO_FIRST,
+)
+
+from tests.common import MockConfigEntry, load_fixture
+
+MOCK_TORRENTS_OLDEST_TO_NEWEST = [2, 1, 3, 4, 5]
+MOCK_TORRENTS_BEST_TO_WORST = [3, 1, 5, 2, 4]
+
+
+@pytest.fixture(name="api")
+def mock_transmission_api():
+    """Mock an api."""
+    with patch("transmissionrpc.Client"):
```

Comment:
Are we patching in the correct place?

https://docs.python.org/3/library/unittest.mock.html#where-to-patch

Reply:
Excellent!  Mocking and patching is becoming demystified ... however this now falls into the category of "how was this ever working" since this is how it had been for years.  It must be that it was getting patched before the import so it works either way.

@MartinHjelmare I would like confirmation that replacing `"transmissionrpc.Client"` with `"homeassistant.components.transmission.transmissionrpc.Client"` is the suggested fix.

Reply:
We're importing the whole library package name, that's why it works. So in this case the existing patch is ok.

https://github.com/home-assistant/core/blob/6f2ed86c49b5721cff4f3678b5bc9e5db7f8c929/homeassistant/components/transmission/__init__.py#L5